### ISSUE TITLE
brew install python leads to continuous error when opening python on Mac OS 10.12

### ISSUE THREAD

**Issue by herryklumper** [Homebrew/brew#1516]
Error processing line 1 of /Library/Python/2.7/site-packages/shapely-py2.7.pth:

  Traceback (most recent call last):
    File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site.py", line 156, in addpackage
      if not dircase in known_paths and os.path.exists(dir):
    File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/genericpath.py", line 26, in exists
      os.stat(path)
  TypeError: stat() argument 1 must be encoded string without null bytes, not str

**Remainder of file ignored**
Error processing line 1 of /Library/Python/2.7/site-packages/sqlite-py2.7.pth:

  Traceback (most recent call last):
    File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site.py", line 156, in addpackage
      if not dircase in known_paths and os.path.exists(dir):
    File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/genericpath.py", line 26, in exists
      os.stat(path)
  TypeError: stat() argument 1 must be encoded string without null bytes, not str

**Remainder of file ignored**
Python 2.7.12 (v2.7.12:d33e0cf91556, Jun 26 2016, 12:10:39) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin

**Comment by mistydemeo** [comment]
It looks like the Python causing an error is one installed in `/Library/Frameworks/Python.framework`, not the one that Homebrew installed.

**Comment by herryklumper** [comment]
Thanks. Good observation. I deleted the Library......\pythons and now python runs normal. Still no pip.

Kindest Regards,

Herry

**Comment by mistydemeo** [comment]
Can you provide the full build logs from your brew Python installation? `brew gist-logs python`

**Comment by herryklumper** [comment]
https://gist.github.com/ca1b772e8673a19da3e31435bc8be9ab

would appreciate a solution.

kindest regards,

herry klumper






> Op 15 nov. 2016, om 23:38 heeft Misty De Meo <notifications@github.com> het volgende geschreven:
> 
> brew gist-logs python

**Comment by mistydemeo** [comment]
It looks like the Python install actually failed, so the Homebrew python was never installed. The error is:

> error: could not delete '/usr/local/lib/python2.7/site-packages/pkg_resources/__init__.py': Permission denied

Try removing the `/usr/local/lib/python2.7` directory, or changing its permissions so your user can write to it, then `brew install python` again.

**Comment by ilovezfs** [comment]
@herryklumper did @mistydemeo's recommendation resolve this for you?

**Comment by herryklumper** [comment]
It did indeed bring me in the right direction. I deleted the directories as indicated and started again with brew install.
Following the directions from brew update and brew doctor (links etc) and then it seems to be working.

Unfortunately it took a long time so was not able to test it yet.

I made one unfortunate mistake and that was that I deleted all files from /usr/local/lib but sofar that did not give a problem.

kind regards,

Herry
